### PR TITLE
Drop CAEFactory

### PR DESCRIPTION
**Pull request by AchimTuran**
To move further with AudioDSP V.2, public API changes for CAEFactory are requiered. It is much better to extend CActiveAE instead of wrapping it with CAEFactory. This will allow us or me to do the chnages directly to CActiveAE instead of CAEFactory.
In my view this improves our current structure a lot, because we can now use CServiceBroker and CServiceManager to get access to CActiveAE.

Feel free to merge @FernetMenta, @fritsch or @AlwinEsch

### CODE REVIEW

File: xbmc/cores/AudioEngine/AEFactory.h [hunk 1]
```diff
@@ -1,83 +0,0 @@
...
-
-  static void RegisterAudioCallback(IAudioCallback* pCallback);
-  static void UnregisterAudioCallback(IAudioCallback* pCallback);
-
-private:
-  static IAE *AE;
-
-  static void SettingOptionsAudioDevicesFillerGeneral(const CSetting *setting, std::vector< std::pair<std::string, std::string> > &list, std::string &current, bool passthrough);
-};
-
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/AEFactory.h]: a single commit should always self-contained and not break the build. otherwise we can't bisect anymore.

Comment (FernetMenta) [inline xbmc/cores/AudioEngine/AEFactory.h]: or is this the last commit in this series? then disregard my comment

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp [hunk 2]
```diff
@@ -2640,6 +2640,33 @@ void CActiveAE::OnSettingsChange(const std::string& setting)
 
 bool CActiveAE::SupportsRaw(AEAudioFormat &format)
 {
+  // check if passthrough is enabled
+  if (!CSettings::GetInstance().GetBool(CSettings::SETTING_AUDIOOUTPUT_PASSTHROUGH))
```
Comment (FernetMenta) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp]: we already have all settings in m_settings: https://github.com/AchimTuran/kodi-agile/blob/3155abd2428ad39fa8f9d9fd1d36a2e4b6909d35/xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp#L2543

lets keep the calls to CSettngs in a central place that is LoadSettngs

**Comment by FernetMenta** [comment]
some minors, see my comments
nice!

File: xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp (hunk 2 above)
Comment (AchimTuran) [inline xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp]: Sorry I didn't see the LoadSettings method.

Should be fixed with lates PR state.

**Comment by AchimTuran** [comment]
Thanks. I hope that all minors are now resolved.

**Pull request merged**